### PR TITLE
[DOC-24] small issues noticed in passing

### PR DESCRIPTION
**Pull request by lazysoundsystem**
The changes to MetadataTrait.php need checking - the rest should be straight-forward.

### CODE REVIEW

File: html/modules/custom/docstore/src/MetadataTrait.php [hunk 1]
```diff
@@ -222,6 +222,7 @@ protected function mapOrCreateTerms($field_name, $values, $type, $provider, $ent
       // Create term.
       $params = [
         'label' => $value,
+        'author' => $provider->uuid(),
```
Comment (attiks) [inline html/modules/custom/docstore/src/MetadataTrait.php:225]: author should be passed in, is different from provider

Comment (attiks) [reply html/modules/custom/docstore/src/MetadataTrait.php:225]: is already passed to `buildItemDataFromMetaData`

File: html/modules/custom/docstore/src/MetadataTrait.php [hunk 2]
```diff
@@ -273,6 +274,7 @@ public function createTermFromParameters(array $params, Vocabulary $vocabulary,
     $item = [
       'name' => $params['label'],
       'vid' => $vocabulary->id(),
+      'author' => [],
```
Comment (attiks) [inline html/modules/custom/docstore/src/MetadataTrait.php:277]: author is required, so will need some more validation

File: html/modules/custom/docstore/src/Controller/VocabularyController.php [hunk 3]
```diff
@@ -924,6 +924,8 @@ public function publishTermRevision($id, $revision_id, Request $request) {
     }
 
     if (!$term->isDefaultRevision()) {
+      // Parse JSON.
+      $params = json_decode($request->getContent(), TRUE);
```
Comment (attiks) [inline html/modules/custom/docstore/src/Controller/VocabularyController.php:928]: Best to move to the top, to be consistent and check.

```
    // Parse JSON.
    $params = json_decode($request->getContent(), TRUE);
    if (empty($params) || !is_array($params)) {
      throw new BadRequestHttpException('You have to pass a JSON object');
    }
```

**Review by attiks — CHANGES_REQUESTED** [review]
Thanks, left some feedback, LMK if I need to make the changes.

Also best to add silk tests for the author on terms

File: html/modules/custom/docstore/src/MetadataTrait.php (hunk 1 above)
Comment (lazysoundsystem) [reply html/modules/custom/docstore/src/MetadataTrait.php:225]: That's what I suspected, but I couldn't see where that's done - e.g. in a lookup in the AR script.

Comment (attiks) [reply html/modules/custom/docstore/src/MetadataTrait.php:225]: buildItemDataFromMetaData gets the author from the params, so pass it to mapOrCreateTerms and it should be fine

File: html/modules/custom/docstore/src/MetadataTrait.php (hunk 2 above)
Comment (lazysoundsystem) [reply html/modules/custom/docstore/src/MetadataTrait.php:277]: I was getting errors that $item['author'] was undefined a few lines on in `$item['author'][] = [`

**Comment by lazysoundsystem** [comment]
Made most of the requested changes, but haven't yet worked out what validation is necessary for author in line 277 of MetadataTrait.php.

**Review by attiks — APPROVED** [review]
I've passed along author and added an extra check

**Comment by lazysoundsystem** [comment]
Thanks!

**Pull request merged**